### PR TITLE
Add support for getting/setting registers on aarch64-linux

### PR DESCRIPTION
**Pull request by s1341**
This PR adds support for getting and setting the general-purpose registers on aarch64-linux-android.

### CODE REVIEW

**Comment by ranweiler** [comment]
Thanks for the PR, @s1341! Aarch64 support is definitely something I've been planning on adding.

At a glance, this looks great. Two quick questions before I fully review it later today:
1. In the PR view, could you please call out any Android (vs. just Aarch64)-specific bits?
1. How did you test this? On what Android/kernel versions?

**Comment by s1341** [comment]
@ranweiler 
1. I don't know if there are any Android-specific bits. I didn't test on anything other than Android. I'm almost certain it should all work on any Aarch64 linux variant. I can try to test that in the next few days.
2. This was tested on Android 9 and Android 11, with various kernel versions. I used pete to attach to a running process, and using `tracee.registers` and `tracee.set_registers`, i injected some function calls.

**Comment by ranweiler** [comment]
@s1341 great! At a glance, it all looked Aarch64-generic to me, too. I'll treat this PR as a prompt to set up ARM integration tests.

File: src/ptracer.rs
Comment (ranweiler) [inline src/ptracer.rs:316]: It'd be nice to share the unsafe code that reads the `user_hwdebug_state`, even just in a private method.

File: src/ptracer.rs
Comment (ranweiler) [reply src/ptracer.rs:275]: Or: return the full `user_hwdebug_state` and leave it to callers to update it appropriately. Since the actual underlying API is just "read the entire debug register context, let the user mess with it, write the entire context", we would typically have `pete` just wrap that way of interacting.

File: src/ptracer.rs
Comment (ranweiler) [inline src/ptracer.rs:303]: As above, I suggest we just have this method write the full debug register struct.

Then we can:
1. Fully reuse `debug_registers()` within this method.
2. Enable callers to efficiently read and write many debug registers at once, with a single syscall sequence.
3. Leave the door open for future Rust wrapper structs that would provide nicer APIs on top of these lower-level methods.

File: src/ptracer.rs
Comment (ranweiler) [reply src/ptracer.rs:189]: (after we merge this PR, I will follow up with non-Android support for Aarch64, for which I have a local test environment)

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs:27]: I will try to open an upstream PR for this.

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs:189]: Doh! why does that discrepancy exist... I have made the `PTRACE_GETREGSET` and `PTRACE_SETREGSET` `i32` on android, and `u32` otherwise.

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs:41]: Fixed.

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs:275]: I've taken the second option.

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs:316]: There is no longer a `GETREGSET` in the `set_debug_registers` function.

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs:303]: Done, except for 1.

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs]: Nope. Just didn't realize it was iovec... Fixed.

File: src/ptracer.rs
Comment (s1341) [reply src/ptracer.rs:29]: https://github.com/rust-lang/libc/pull/2316

File: src/ptracer.rs
Comment (ranweiler) [inline src/ptracer.rs:492]: Is this related to the rest of the PR? What's motivating this change?

Comment (s1341) [reply src/ptracer.rs:492]: Woops. That slipped in. It's not related to the rest of the PR. I'll excise it. It is important and I'll open a separate PR for this. It makes it possible to catch `SIGINT`s. Without this, they are swallowed up.

File: src/ptracer.rs
Comment (ranweiler) [inline src/ptracer.rs:305]:
```suggestion
```

**Review by ranweiler — APPROVED** [review]
Thanks very much for this, @s1341! I'll merge this tonight after a final (manual) integration test.

**Comment by ranweiler** [comment]
Thanks a ton for this, @s1341! I'm going to follow up with some maintenance PRs, and will then include this in a new release.

**Comment by s1341** [comment]
Thanks for merging!

I will do a follow-up PR for the `SIGINT` wait issue. Also, my PR on `libc` was merged.

**Pull request merged**
